errors.go

```diff
@@ -282,7 +282,11 @@ func Cause(err error) error {
 		if !ok {
 			break
 		}
-		err = cause.Cause()
+		if cerr := cause.Cause(); cerr != nil {
```

Comment:
We can avoid the spurious `break` control flow by structuring this the other way around:
```
cerr := cause.Cause()
if cerr == nil {
	return err
}
err = cerr
```